test/test_shell.py

```diff
@@ -156,6 +157,42 @@ async def check_output(cmd, **kwargs):
     assert 'DECODE_ERROR=' in warn.call_args[0][0]
 
 
+def test_get_null_separated_environment_variables():
+    cmd = [
+        sys.executable, '-c',
+        r'print("FOO\0NAME=value\nSOMETHING\0NAME2=value with spaces'
+        r'\0NAME3=NAME4\nNAME5=NAME6")']
+
+    coroutine = get_null_separated_environment_variables(cmd, shell=False)
+    env = run_until_complete(coroutine)
+
+    assert len(env.keys()) == 3
+    assert 'NAME' in env.keys()
+    assert env['NAME'] == f'value{os.linesep}SOMETHING'
```

Comment:
Why is this using `os.linesep` here, but `\n` in the input data above? The test in https://github.com/colcon/colcon-core/pull/128 tested `assert env['NAME'] == 'value\nSOMETHING'`. Wouldn't this break for Windows with `\r\n` as line break?

Reply:
The subprocess that does the `print` was normalizing newlines on stdout. Rather than compensating for that on the "other side", I can change the subprocess to print raw bytes.